api/src/main/java/org/geysermc/geyser/api/connection/GeyserConnection.java

```diff
@@ -95,4 +98,12 @@ public interface GeyserConnection extends Connection, CommandSource {
      */
     @NonNull
     Set<String> fogEffects();
+
+    <T> void storePreference(@NonNull PreferenceKey<T> key, @NonNull Preference<T> preference);
```

Comment:
Javadocs here please. We should emphasize this system is Geyser and not Bedrock.